src/modules/folder/folder.usecase.ts

```diff
@@ -294,12 +294,18 @@ export class FolderUseCases {
   async moveFoldersToTrash(
     user: User,
     folderIds: FolderAttributes['id'][],
+    folderUuids: FolderAttributes['uuid'][] = [],
   ): Promise<void> {
-    const [folders, driveRootFolder] = await Promise.all([
+    const [foldersById, driveRootFolder, foldersByUuid] = await Promise.all([
       this.getFoldersByIds(user, folderIds),
       this.getFolder(user.rootFolderId),
+      folderUuids.length > 0
```

Comment:
Hey, we do not want to move items that already trashed into the trash again, therefore, you need to add:
`where deleted = "false" and removed = "false"`
So you can prevent concurrency issues if multiples devices are syncing.

Reply:
Gotcha, added the conditions to the repository so trashed  folders are not fetch.